### PR TITLE
This is for code review only - NOT FOR MERGE (Related to RavenDB-12503, RavenDB-12206 and RavenDB-12187)

### PR DESCRIPTION
**Pull request by myarichuk**
1) Make sure that filters & select in edge will work when there is string array as edge property
2) Add unit test for RavenDB-12187 - skipped for now, will be "unskipped" as the issue is fixed
3) Unskip & fix several unit tests that were skipped before (they were skipped because of unimplemented issues)

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs [hunk 1]
```diff
@@ -44,27 +45,59 @@ public void SingleMatch(Match left, string alias)
...
+                            switch (item)
                             {
-                                AddEdgeAfterFiltering(left, json, Edge.Project?.FieldValue);
+                                case BlittableJsonReaderObject json when Edge.Where?.IsMatchedBy(json, QueryParameters) != false:
+                                    if (projectFieldValue == null)
+                                        ThrowMissingEdgeProjection();
+                                    AddEdgeAfterFiltering(left, json, projectFieldValue);
+                                    break;
+                                case LazyStringValue lazyString:
+                                    var jsonStringForWhere = new DynamicJsonValue
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs]: Don't we need to check that `projectFieldValue` is null here?

Comment (myarichuk) [reply src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs]: Probably (will need to test if it can become null and how), and also we need to test for arrays of primitives that are not strings.
I did this PR because I am not 100% sure that the code so far is good enough - and I mean the code around IsMatchedBy() that I added

**Comment by myarichuk** [comment]
added another commit, better handling of use-cases for RavenDB-12206
(planning to submit separate PR for RavenDB-12206, as soon as I synchronize with Tal's code changes)

**Comment by myarichuk** [comment]
rewrote the commit, make sure all tests are passing

**Comment by myarichuk** [comment]
will submit stuff as per-issue PRs
